spec/controllers/categories_controller_spec.rb

```diff
@@ -1,4 +1,33 @@
 require 'rails_helper'
 
 RSpec.describe CategoriesController, type: :controller do
+  let(:admin) { create(:admin) }
+  let(:category) { create(:category) }
+
+  before(:each) do
+    @request.env["devise.mapping"] = Devise.mappings[:user]
+    sign_in admin
+  end
+
+  describe "GET #new" do
+    it "assigns a new category as @category" do
+      get :new
+      expect(assigns(:category)).to be_a_new(Category)
+    end
+  end
+
+  describe "POST create" do
+    context "with valid attributes" do
```

Comment:
Thumbs up for using a context with valid params for succession test.

try to add a context with invalid params for failing test too. tia!